assets/model_monitoring/components/src/action_analyzer/contracts/utils/detector_utils.py

```diff
@@ -0,0 +1,406 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""This file contains utilities for action detector."""
+import os
+import uuid
+import json
+import pandas
+import yaml
+import requests
+import re
+from typing import List
+from mlflow import MlflowClient
+from shared_utilities.span_tree_utils import SpanTree
+from shared_utilities.constants import (
+    RETRIEVAL_SPAN_TYPE,
+    GSQ_METRICS_LIST,
+    MODIFIED_PROMPT_COLUMN,
+    COMPLETION_COLUMN,
+    RETRIEVAL_DOC_COLUMN,
+    SPAN_ID_COLUMN,
+    INDEX_CONTENT_COLUMN,
+    INDEX_ID_COLUMN,
+    INDEX_SCORE_COLUMN,
+    RETRIEVAL_QUERY_TYPE_COLUMN,
+    RETRIEVAL_TOP_K_COLUMN,
+    PROMPT_FLOW_INPUT_COLUMN,
+    INVALID_LLM_SCORE,
+    ROOT_SPAN_COLUMN,
+    INDEX_SCORE_LLM_COLUMN,
+    PROMPT_COLUMN,
+    DEFAULT_TOPIC_NAME,
+    TEXT_SPLITTER,
+    MAX_SAMPLE_SIZE
+)
+from shared_utilities.llm_utils import _APITokenManager, _request_api
+from shared_utilities.io_utils import (
+    np_encoder
+)
+from shared_utilities.amlfs import amlfs_upload
+from shared_utilities.prompts import RELEVANCE_TEMPLATE, QUERY_INTENTION_PROMPT
+from action_analyzer.contracts.action_sample import IndexActionSample
+from action_analyzer.contracts.actions.action import Action
+from action_analyzer.contracts.llm_client import LLMClient
+
+
+def get_index_id_from_index_content(index_content: str) -> str:
+    """Parse the index id from index yaml.
+
+    Args:
+        index_content(str): str of index yaml content.
+
+    Returns:
+        str: the index id parsed from the index yaml content.
+    """
+    index_payload = yaml.safe_load(index_content)
+    # if the asset id does not exist, use the index name
+    if "self" in index_payload:
```

Comment:
An can handle the data, right? index_id can be assetid or other string?

Reply:
They cannot handle. We can discuss this tomorrow.